flux-middle/src/fhir.rs

```diff
@@ -150,6 +150,29 @@ pub enum RefKind {
     Mut,
 }
 
+#[derive(PartialEq, Eq, PartialOrd, Ord, Hash, Clone, Copy)]
+pub enum WeakKind {
+    Shr,
+    Mut,
+    Arr,
+}
+
+impl WeakKind {
+    pub fn from_ref_kind(rk: RefKind) -> WeakKind {
+        match rk {
+            RefKind::Shr => WeakKind::Shr,
+            RefKind::Mut => WeakKind::Mut,
+        }
+    }
```

Comment:
The idiomatic way is to `impl From<RefKind> for WeaKind`

Reply:
aha, thanks for the tip!